src/eureka/S4_generate_lightcurves/generate_LD.py

```diff
@@ -81,8 +81,22 @@ def exotic_ld(meta, spec, log, white=False):
         wavelength_range *= 1e4
 
     # compute stellar limb darkening model
-    sld = StellarLimbDarkening(meta.metallicity, meta.teff, meta.logg,
-                               meta.exotic_ld_grid, meta.exotic_ld_direc)
+    if hasattr(meta, "custom_si_grid") and meta.exotic_ld_grid == 'custom':
+        s_wvs = (np.genfromtxt(meta.custom_si_grid, 
+                               skip_header=2, usecols=[0]).T)*1e4
+        s_mus = np.flip(np.genfromtxt(meta.custom_si_grid, 
+                                      skip_header=1, max_rows=1))
+        custom_si = np.flip(np.genfromtxt(meta.custom_si_grid, 
+                                          skip_header=2)[:, 1:], axis=1)
```

Comment:
Why load the file into memory three separate times? Would seem much more efficient to load it once. There is also some nuanced processing here that is not easily understandable to me at first glance, which is going to make the code hard to maintain. Can you add some comments here to explain what is happening? What is the expected formatting of the file (should also be noted in the docs), why do you sometimes use `np.flip` but not always, why do you sometimes set `skip_header` to 1 and other times 2, etc.

Reply:
It's read in three times because you're grabbing different parts of the file each time - but I can read it in once and just trim out the necessary parts for each input if you'd prefer. The file is a table with a 1 line header, then the first column is the wavelengths and the first row is the mus. Exotic requires arrays of wavelength and mu separate from the 2D array of intensities. So when grabbing the mus, you have to skip the one row header and only read the first row, but for the wavelengths and intensities you skip the header and the row of the mu values.  And mu has to be decreasing for exotic, whereas intensity files are typically in increasing mu (hence the flip for those). I'll add some notes into here on what is being done.

Reply:
Reading it in once would speed things up (since I/O tends to be one of the slowest parts of programs), but its not worth slowing down this PR for since your code works and its not like this is done often and the files aren't huge. Thanks for the description and the added comments in the code